### PR TITLE
Add dynamo expand test. 

### PR DESCRIPTION
**Pull request by ysiraichi**
This PR adds a test for #5837. The fix was introduced in PyTorch main repository (https://github.com/pytorch/pytorch/pull/121007), but we need PyTorch/XLA for actually exercising the (previously) failing test case.

cc @miladm @JackCaoG

### CODE REVIEW

File: test/dynamo/test_dynamo.py [hunk 1]
```diff
@@ -568,6 +568,24 @@ def test_all_cpu_tensor(self):
     self.assertIn('MarkStep', met.counter_names())
 
 
+class DynamoOperationsTests(test_utils.XlaTestCase):
```
Comment (miladm) [inline test/dynamo/test_dynamo.py:571]: can we add a flag to skip this test while we address the corresponding bug you will soon open on "expand  failing on dynamo"

**Comment by alanwaketan** [comment]
You can add a .torch_pin to build with your patch PR. Following the instructions here: https://github.com/pytorch/xla/tree/master/torch_patches

Such that we don't need to add any guards. We just need to land the two patches together.

**Comment by alanwaketan** [comment]
Great, looks like all tests pass! Please ping me once your upstream PR is approved. Also, we need to remove the .torch_pin before landing the PR.

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Pull request merged**